internal/indexer/document_change.go

```diff
@@ -124,6 +124,19 @@ func (idx *Indexer) decodeModule(ctx context.Context, modHandle document.DirHand
 	}
 	ids = append(ids, refOriginsId)
 
+	_, err = idx.jobStore.EnqueueJob(ctx, job.Job{
```

Comment:
Is there a thought process behind why we should or should not wait for this job to finish? We seem to wait for all of them except for the one below.

I believe the original idea was that anything that would be affected by the change (such as parsing) and where it matters that we process changes in the right order, we should wait. Technically the client won't _really_ wait in the sense that the UI shouldn't be blocked but it may affect how we process next change notifications.

Here I think we should try to avoid publishing outdated diagnostics.

TL;DR I think we should wait, i.e. return job ID and append to the slice here.

Reply:
We can observe the UX to later understand the impact of that. I suppose most of the concerns I have about outdated diagnostics could be alleviated by us using the document `version` in our requests

https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#publishDiagnosticsParams

```ts
	/**
	 * Optional the version number of the document the diagnostics are published
	 * for.
	 *
	 * @since 3.15.0
	 */
	version?: [integer](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#integer);
```

It still seems wasteful to be computing diagnostics for a version which is already outdated by the time we compute them but maybe that's a different problem that we can't really address here anyway.

So TL;DR feel free to merge as-is, we can look into it later if necessary.